services/signature.proto

```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "basic_types.proto";
+
+/**
+ * The record signature file which is created for each record stream file
+ * that signs the hash of the entire corresponding stream file.
+ */
+message SignatureFile {
+
+  /**
+   * Signature for the file
+   */
+  SignatureObject fileSignature = 1;
+
+  /**
+   * Metadata signature
+   */
+  SignatureObject metadataSignature = 2;
+}
+
+/**
+ * A Signature defined by its type, length, checksum and signature bytes and the hash that is signed
+ */
+message SignatureObject {
+
+  /**
+   * The signature type
+   */
+  enum SignatureType {
+    SHA_384_WITH_RSA = 0;
```

Comment:
Is this really considered field=0 within `SignatureObject`? Maybe we should move its definition outside of `SignatureObject` so `type = 0`?

Reply:
I moved the definitions of the enum outside the messages, but still the type cannot be 0.

https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers

Moreover, there is an error "test.proto:2:28: Field numbers must be positive integers." when trying to assign the field to 0.